src/server/transaction.cc

```diff
@@ -1160,24 +1163,16 @@ OpStatus Transaction::WatchInShard(ArgSlice keys, EngineShard* shard) {
   return OpStatus::OK;
 }
 
-void Transaction::UnwatchShardCb(ArgSlice wkeys, bool should_expire, EngineShard* shard) {
-  if (should_expire) {
-    auto lock_args = GetLockArgs(shard->shard_id());
-    shard->db_slice().Release(Mode(), lock_args);
-
-    unsigned sd_idx = SidToId(shard->shard_id());
-    auto& sd = shard_data_[sd_idx];
-    sd.local_mask |= EXPIRED_Q;
-    sd.local_mask &= ~KEYLOCK_ACQUIRED;
-    shard->blocking_controller()->FinalizeWatched(wkeys, this);
-    DCHECK(!shard->blocking_controller()->awakened_transactions().contains(this));
-  }
+void Transaction::ExpireBlockingShardCb(ArgSlice wkeys, EngineShard* shard) {
```

Comment:
I think this PR is not synced with the latest fixes. Can wait for your return for sure.